int16 limit_number
bool limit_start
